scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -404,12 +404,12 @@ class Router(formatOps: FormatOps) {
             style.newlines.afterCurlyLambdaParams ne
               Newlines.AfterCurlyLambdaParams.squash
 
-          isCurlyLambda && (style.newlines.source match {
+          style.newlines.source match {
```

Comment:
i think fold case should replace false with isCurlyLamvda, and keep should become newlines != 0 || isCurlyLambda.

Reply:
That seems to break a bunch of tests:
```
==> X org.scalafmt.FormatTests.newlines/source_fold.stat:1554: preserve ctor body with zero args     |  0.004s munit.ComparisonFailException: /home/tgodzik/Documents/scalafmt/scalafmt-tests/src/test/resources/newlines/source_fold.stat:1554
1553:val foo = new Bar { () => f(42) }
1554:<<< preserve ctor body with zero args
1555:val foo = new Bar { () => f(42) }
values are not the same
=> Obtained
val foo = new Bar { () =>
  f(42)
}

=> Diff (- obtained, + expected)
-val foo = new Bar { () =>
-  f(42)
-}
+val foo = new Bar { () => f(42) }
```

Looks like fold needs to stay false. As for `newlines != 0 || isCurlyLambda.` that breaks:

```
==> X org.scalafmt.FormatTests.newlines/source_keep.stat:1656: keep ctor body with zero args         |  0.027s munit.ComparisonFailException: /home/tgodzik/Documents/scalafmt/scalafmt-tests/src/test/resources/newlines/source_keep.stat:1656
1655:}
1656:<<< keep ctor body with zero args
1657:val foo = new Bar { () => f(42) }
values are not the same
=> Obtained
val foo = new Bar { () =>
  f(42)
}

=> Diff (- obtained, + expected)
-val foo = new Bar { () =>
-  f(42)
-}
+val foo = new Bar { () => f(42) }
```

Reply:
yes, you're right, i shouldn't review code in the middle of the night. the old logic was isCurlyLambda && false, which is false.